airflow/jobs/scheduler_job.py

```diff
@@ -739,18 +738,11 @@ def _do_scheduling(self, session) -> int:
 
             callback_tuples = []
             for dag_run in dag_runs:
-                # Use try_except to not stop the Scheduler when a Serialized DAG is not found
-                # This takes care of Dynamic DAGs especially
-                # SerializedDagNotFound should not happen here in the same loop because the DagRun would
-                # not be created in self._create_dag_runs if Serialized DAG does not exist
-                # But this would take care of the scenario when the Scheduler is restarted after DagRun is
-                # created and the DAG is deleted / renamed
-                try:
-                    callback_to_run = self._schedule_dag_run(dag_run, session)
-                    callback_tuples.append((dag_run, callback_to_run))
-                except SerializedDagNotFound:
-                    self.log.exception("DAG '%s' not found in serialized_dag table", dag_run.dag_id)
+                # _schedule_dag_run will return 0 if SerializedDag is not found
+                callback_to_run = self._schedule_dag_run(dag_run, session)
+                if callback_to_run == 0:
```

Comment:
Returning 0 is a separate bug, but we don't care if it's zero or a callback here -- just a "false value" is enough for us here.

```suggestion
                callback_to_run = self._schedule_dag_run(dag_run, session)
                if not callback_to_run:
```